libnetwork/network.go

```diff
@@ -890,46 +891,31 @@ func NetworkDeleteOptionRemoveLB(p *networkDeleteParams) {
 	p.rmLBEndpoint = true
 }
 
-func (n *Network) resolveDriver(name string, load bool) (driverapi.Driver, driverapi.Capability, error) {
-	c := n.getController()
-
-	// Check if a driver for the specified network type is available
-	d, capabilities := c.drvRegistry.Driver(name)
-	if d == nil {
-		if load {
-			err := c.loadDriver(name)
-			if err != nil {
-				return nil, driverapi.Capability{}, err
-			}
-
-			d, capabilities = c.drvRegistry.Driver(name)
-			if d == nil {
-				return nil, driverapi.Capability{}, fmt.Errorf("could not resolve driver %s in registry", name)
-			}
-		} else {
-			// don't fail if driver loading is not required
-			return nil, driverapi.Capability{}, nil
-		}
-	}
-
-	return d, capabilities, nil
-}
-
 func (n *Network) driverIsMultihost() bool {
-	_, capabilities, err := n.resolveDriver(n.networkType, true)
+	_, err := n.driver(true)
 	if err != nil {
 		return false
 	}
-	return capabilities.ConnectivityScope == scope.Global
+	return n.driverCaps.ConnectivityScope == scope.Global
 }
 
+// driver returns the network [driverapi.Driver] used by this Network. If load
+// is false and the appropriate driver can't be found, it will return a nil
+// driver and no error.
 func (n *Network) driver(load bool) (driverapi.Driver, error) {
-	d, capabilities, err := n.resolveDriver(n.networkType, load)
+	d, capabilities, err := n.ctrlr.resolveDriver(n.networkType, load)
 	if err != nil {
 		return nil, err
 	}
 
 	n.mu.Lock()
+	// n.driverCaps is initialized here for two reasons: 1. resolveDriver is
+	// called early by NewNetwork; 2. networks restored after introducing this
+	// new field need to have it set.
+	// TODO(aker): move this to NewNetwork once the next MCR LTS is published.
```

Comment:
`n.driverCaps` isn't included in the JSON marshalled `Network` (or its `CopyTo`). It needs to be if we want to move this to NewNetwork?